docs/runtimes/image-build.md

```diff
@@ -0,0 +1,192 @@
+(build-function-image)=
+# Build function image
+
+As discussed in {ref}`images-usage`, MLRun provides pre-built images which contain the components necessary to execute
+an MLRun runtime. In some cases, however, custom images need to be created. 
+This page details this process and the available options.
+
+## When is a build required?
+
+In many cases an MLRun runtime can be executed without having to build an image. This will be true when
+the basic MLRun images fulfill all the requirements for the code to execute. It will be required to build an image 
+if one of the following is true:
+
+1. Code is using additional Python packages, OS packages, scripts or other configurations that need to be applied
+2. Using different base-images or different versions of MLRun images than provided by default
+3. Executed source code has changed, and the image has the code packaged in it - see
+   [here](mlrun_jobs.html#deploy-build-the-function-container) for more details on source code, and using 
+   {py:func}`~mlrun.runtimes.BaseRuntime.with_code()` to avoid re-building the image when the code has changed
+4. Running nuclio functions, which are packaged as images (build is triggered by MLRun and performed by nuclio)
+
+The build process in MLRun is based on [Kaniko](https://github.com/GoogleContainerTools/kaniko) and automated by MLRun -
+MLRun generates the dockerfile for the build process, and configures Kaniko with parameters needed for the build.
+
+Building images is done through functions provided by the {py:class}`~mlrun.projects.MlrunProject` class. By using 
+project functions, the same process is used to build and deploy a stand-alone function or functions serving as steps 
+in a pipeline.
+
+## Automatically building images
+MLRun has the capability to auto-detect when a function image needs to first be built. Following is an example that
+will require building of the image:
+
+```python
+project = mlrun.new_project(project_name, "./proj")
+
+project.set_function(
+   "train_code.py", 
+   name="trainer",
+   kind="job",
+   image="mlrun/mlrun",
+   handler="train_func",
+   requirements=["pandas"]
+)
+
+# auto_build will trigger building the image before running, 
+# due to the additional requirements.
+project.run_function("trainer", auto_build=True)
+```
+
+Using the `auto_build` option is only suitable when the build configuration does not change between runs of the
+runtime. For example, if during the development process new requirements were added, the `auto_build` parameter should
+not be used, and manual build will be needed to re-trigger a build of the image.
+
+In the example above, the `requirements` parameter was used to specify a list of additional Python packages required by
+the code. This option directly affects the image build process - each requirement will be installed using `pip` as 
+part of the docker-build process. The `requirements` parameter can also contain a path to a requirements file, making
```

Comment:
This option directly affects the image build process - each requirement is installed using `pip` as 
part of the docker-build process.